mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java

```diff
@@ -0,0 +1,51 @@
+package com.github.cloudyrock.spring.v5;
```

Comment:
As you mentioned, it doesn't prove the bean is not injected due to enabled=false.
We'd need to prove the base condition, meaning that the beans are injected when enabled is true or missing.

Reply:
What you would be testing then in a way would then be the effectiveness of the @Conditional annotation, which I am sure the Spring team has done extensively.

Due to the hard dependencies of the beans to MongoTemplate, I would need to stand up a full App Context (and likely test containers) to have a working e2e test. Is this what you would like?

Reply:
Maybe I am missing something here, butifarraI don't see why you need TestContainers. As we are not performing any database action or check, we can mock it.

The only issue I think you could have is related to both drivers, mongodb-springdata-v3-driver and mongodb-springdata-v2-driver, as you would probably need to import one of them.

If it's too painful, don't worry, I do it. we have a separate project, one per driver, for integration tests.
But let me know